src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -230,14 +230,17 @@ private[internal] trait GlbLubs {
   def weakLub(tps: List[Type]): Type = (
     if (tps.isEmpty)
       NothingTpe
-    else if (tps forall isNumericValueType)
+    else if (shouldUseNumericLub(tps))
       numericLub(tps)
     else if (tps exists typeHasAnnotations)
       annotationsLub(lub(tps map (_.withoutAnnotations)), tps)
     else
       lub(tps)
   )
 
+  private def shouldUseNumericLub(tps: List[Type]) =
+    tps forall (tp => isNumericValueType(tp.widen))
```

Comment:
I quite nearly just put the `.widen` in `isNumericValueType`, which doesn't appear to break anything. It's only used (in the compiler) here and [in `TypeConstraints`](https://github.com/scala/scala/blob/f6e46eb2b2c84d2fc8ac26f8d400004e99aa43be/src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala#L136), and since I haven't yet found a way to get there with a numeric `ConstantType` I can't convince myself it doesn't cause breakage. Also then I'd arguably have to rename the method and I'm running out of my names budget for the week.